src/Controller/Hello.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class Hello
+{
+    use ControllerTrait;
+
+    /**
+     * @Route("/", name="homepage")
+     */
+    public function __invoke(Request $request): Response
```

Comment:
@yceruto Indeed, it does not involve magic call, but as said, it _helps_ to realize that what Symfony calls a _controller_ is actually _any PHP callable_. Somehow this concept is not totally obvious for everyone, and as said above, @dunglas proposes ADR pattern, so using an invokable class is basically the best solution for this, even using the `@Route` annotation.
 
And I'd personally get rid of the annotation in favor of yml routing :trollface: